### ISSUE TITLE
Accessibility feedback for Design intent Collaboration Facilities [Map Enhancements: Phase 1]

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#15349]
**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
 
- **✔️ Consider** for suggestions/enhancements 

## Point of Contact

VSA: Jennifer
VSP: Trevor

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

### Questions

1. Do you know if the VA is required to meet any different regulations regarding asking for permission to geolocate automatically? Also, was this something that any Vets were ever asked if what they thought of allowing VA.gov to geolocate them? 

### Must

1. For the custom component to try adding a paragraph to the 'no results' option, the new component **must** be tested with all devices, browsers, and screen reader combinations. Custom components require a lot more and extra work than the default code elements. This is not intended to discourage, though. It sounds like a great idea, that those with assistive tech would also want access to.

1. For the screen reader experience, offering suggestions at one character is a pretty challenging experience, particularly due to the web performance issues on the Facility Locator. There's a delay, so the screen reader user hears a letter, then the beginning of the pronunciation of the word being formed. As the work progresses, it **must** include screen reader testing as part of the process.

1. The 'Search this area' solution **must** make screen reader users aware of the introduction of the button, and then manage focus to the new search results list. 
    1. For the introduction of the "Search this area" button, zooming or panning the map *must* announce the "Search this area" button for screen reader users (the `aria-live` attribute is likely to be helpful; it depends on how this is built). 
    1. When a user activates the button to search this area, focus **must** be set on the heading for `Results for "VA health", "All VA health services" near "Arlington, Virginia 22202"` (with the actual criteria, of course). That way screen reader users will hear the new search results, along with the new geographic location.
    1. Managing focus / announcing the "Search this area" button is a little trickier. We should announce it when the button appears, but we have two distinct paradigms to consider:
        1. Users pan by dragging with a finger or pointer device. If we don't manage focus to the button, users might have to swipe or press TAB a number of times to reach the "Search this area" button.
        1. Users press the ` + ` or ` - ` button to zoom the map. We need to communicate the button is now on the screen if it wasn't before, but do we move the user focus? My thought is maybe. If we always inject the button just after the zoom in/zoom out buttons in the source order, can we announce the button + an instructional text that says to "Tab or swipe right twice, then click or tap twice to search"? This feels better than putting the button at the end of the source order and moving focus dynamically.
    1. Testing with multiple screen reader users would be helpful to ensure the solution isn't disorienting.
    1. These two paradigms will probably take an iterative approach to get the UX right. That's alright, this is a super cool feature!

### Consider 

1. **Consider** creating specifications to document the focus management (your accessibility specialist can assist), which will make development easier.

**Comment by 1Copenut** [comment]
@jenstrickland I agree with the assessment that zooming or panning the map *must* announce the "Search this area" button. We should set focus on the *Results for "VA Health" near "Washington, D.C."* when users press the button. That will trigger screen readers to announce the new result set with a geographic location.

Managing focus / announcing the "Search this area" button is a little trickier. We should announce it when the button appears, but we have two distinct paradigms to consider:

1. Users pan by dragging with a finger or pointer device. If we don't manage focus to the button, users might have to swipe or press TAB a number of times to reach the "Search this area" button.
2. Users press the ` + ` or ` - ` button to zoom the map. We need to communicate the button is now on the screen if it wasn't before, but do we move the user focus? My thought is maybe. If we always inject the button just after the zoom in/zoom out buttons in the source order, can we announce the button + an instructional text that says to "Tab or swipe right twice, then click or tap twice to search"? This feels better than putting the button at the end of the source order and moving focus dynamically.

These two paradigms will probably take an iterative approach to get the UX right. That's alright, this is a super cool feature!

**Comment by mmiddaugh** [comment]
Focus management has been reviewed with Josh and Trevor and is either resolved or in progress
